src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +109,845 @@ public static void init()
+        final File schematicsFolder = Structure.getSchematicsFolder();
+        if (schematicsFolder != null)
+        {
+            Log.getLogger().info("Load additionnal huts or decorations from " + schematicsFolder + SCHEMATICS_SEPARATOR + SCHEMATICS_PREFIX);
+            checkDirectory(schematicsFolder.toPath().resolve(SCHEMATICS_PREFIX).toFile());
+            loadSchematicsForPrefix(schematicsFolder.toPath(), SCHEMATICS_PREFIX);
+        }
 
-            if (FMLCommonHandler.instance().getMinecraftServerInstance() == null)
-            {
-                decorationFolder = new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/decorations");
-            }
-            else
-            {
-                decorationFolder = new File(FMLCommonHandler.instance().getMinecraftServerInstance().getDataDirectory(), "minecolonies/decorations");
-            }
+        final File cacheSchematicFolder = Structure.getCachedSchematicsFolder();
+        if (cacheSchematicFolder != null)
+        {
+            checkDirectory(cacheSchematicFolder);
+            Log.getLogger().info("Load cached schematic from " + cacheSchematicFolder + SCHEMATICS_SEPARATOR + SCHEMATICS_CACHE);
+            loadSchematicsForPrefix(cacheSchematicFolder.toPath(), SCHEMATICS_CACHE);
+        }
 
-            if (!decorationFolder.exists() && !decorationFolder.mkdirs())
-            {
-                Log.getLogger().warn("Failed to create directories for dynamic decorations.");
-            }
-            loadStyleMaps(decorationFolder.toPath());
+        if (md5Map.size() == 0)
+        {
+            Log.getLogger().error("Error loading StructureProxy directory. Things will break!");
         }
-        catch (@NotNull IOException | URISyntaxException e)
+    }
+
+    /**
+     * Load all schematics from the scan folder.
+     */
+    @SideOnly(Side.CLIENT)
+    public static void loadScannedStyleMaps()
+    {
+        if (!allowPlayerSchematics && FMLCommonHandler.instance().getMinecraftServerInstance() == null)
         {
-            Log.getLogger().error("Error loading StructureProxy directory. Things will break!", e);
+            return;
         }
+
+
+        schematicsMap.remove(SCHEMATICS_SCAN);
+        final File schematicsFolder = Structure.getClientSchematicsFolder();
+        checkDirectory(schematicsFolder.toPath().resolve(SCHEMATICS_SCAN).toFile());
+        loadSchematicsForPrefix(schematicsFolder.toPath(), SCHEMATICS_SCAN);
     }
 
     /**
      * Load all style maps from a certain path.
+     * load all the schematics inside the folder path/prefix
+     * and add them in the md5Map
      *
      * @param basePath the base path.
-     * @throws IOException if nothing found.
+     * @param prefix   either schematics, scans, cache
      */
-    public static void loadStyleMaps(final Path basePath) throws IOException
+    private static void loadSchematicsForPrefix(@NotNull final Path basePath, @NotNull final String prefix)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)